cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/dataset/Metadata.java

```diff
@@ -27,19 +27,19 @@
 /**
  * Represents the complete metadata of a {@link Id.NamespacedId} including its properties and tags.
  */
-public class MetadataHistoryEntry {
+public class Metadata {
```

Comment:
Today MetadataEntry means a key and a value, and Metadata means the whole metadata for the entity - properties and tags. 

We need more re-factoring in metadata codebase, it appears that MetadataEntry is redundant. We may remove it in next refactoring. Even then I think this class name should still be Metadata, since it is the metadata for the entity. What do you say?